chef.py

```diff
@@ -0,0 +1,156 @@
+            get_subtopics(node, topic['href'])
+            if DEBUG_MODE:
+                return
+        except Exception as e:
+            LOGGER.error('get_topics: %s : %s' % (e, topic))
+
+
+def get_subtopics(parent, path):
+    doc = get_page(path)
+    try:
+        menu_row = doc.find('div', {'id': 'body-row'})
+        menu_row = menu_row.find('div', {'class': 'col-md-2'})
+    except Exception as e:
+        LOGGER.error('get_subtopics: %s : %s' % (e, doc))
+    for subtopic in menu_row.find_all('a'):
+        try:
+            LOGGER.info('subtopic: %s' % (subtopic['href']))
+            title = subtopic.get_text().strip()
+            source_id = get_source_id(subtopic['href'])
+            node = TopicNode(title=title, source_id=source_id)
+            parent.add_child(node)
+            get_lessons(node, subtopic['href'])
+        except Exception as e:
+            LOGGER.error('get_subtopics: %s : %s' % (e, subtopic))
+
+
+def get_lessons(parent, path):
+    doc = get_page(path)
+    try:
+        menu_row = doc.find('div', {'id': 'body-row'})
+        menu_row = menu_row.find('div', {'class': 'col-md-9'})
+    except Exception as e:
+        LOGGER.error('get_lessons: %s : %s' % (e, doc))
+    for lesson in menu_row.find_all('div', {'class': 'thumbnail'}):
+        try:
+            title = lesson.find('div', {'class': 'txtline'}).get_text().strip()
+            link = lesson.find('a')['href']
+            LOGGER.info('lesson: %s' % (link))
+            source_id = get_source_id(link)
+            node = TopicNode(title=title, source_id=source_id)
+            parent.add_child(node)
+            get_contents(node, link)
+        except Exception as e:
+            LOGGER.error('get_lessons: %s : %s' % (e, lesson))
+
+
+def get_contents(parent, path):
+    doc = get_page(path)
+    try:
+        menu_row = doc.find('div', {'id': 'row-exu'})
+    except Exception as e:
+        LOGGER.error('get_contents: %s : %s' % (e, doc))
+    for content in menu_row.find_all('div', {'class': 'col-md-3'}):
+        try:
+            title = content.find('div', {'class': 'txtline'}).get_text()
+            link = content.find('a', {'title': 'Download'})
+            if link:
+                # This link is relative to the current page.
+                link = urllib.parse.urljoin(get_absolute_path(path), link['href'])
+            else:
+                # let's get it from the onclick attribute
+                link = content.find('a', {'id': 'navigate'})
+                regex = re.compile(r"res_click\('(.*)','.*','.*','.*'\)")
+                match = regex.search(link['onclick'])
+                link = match.group(1)
+                link = get_absolute_path(link)
+            if link.endswith('mp4'):
+                video = VideoNode(
+                    title=title,
+                    source_id=get_source_id(link),
+                    license=licenses.PUBLIC_DOMAIN,
+                    files=[VideoFile(link)])
```

Comment:
Open question: is the existing resolution/bitrate good, or do we want to pass compression settings here?